pyaedt/edb_core/edb_data/siwave_simulation_setup_data.py

```diff
@@ -1,19 +1,131 @@
-from pyaedt.edb_core.edb_data.hfss_simulation_setup_data import EdbFrequencySweep
+from pyaedt.edb_core.edb_data.simulation_setup import BaseSimulationSetup
 from pyaedt.edb_core.general import convert_netdict_to_pydict
 from pyaedt.edb_core.general import convert_pydict_to_netdict
-from pyaedt.generic.general_methods import generate_unique_name
 from pyaedt.generic.general_methods import is_linux
 from pyaedt.generic.general_methods import pyaedt_function_handler
 
 
-class SiwaveAdvancedSettings(object):
+def _parse_value(v):
+    """Parse value in C sharp format."""
+    #  duck typing parse of the value 'v'
+    if v is None or v == "":
+        pv = v
+    elif v == "true":
+        pv = True
+    elif v == "false":
+        pv = False
+    else:
+        try:
+            pv = int(v)
+        except ValueError:
+            try:
+                pv = float(v)
+            except ValueError:
+                if isinstance(v, str) and v[0] == v[-1] == "'":
+                    pv = v[1:-1]
+                else:
+                    pv = v
+    return pv
+
+
+class SettingsBase(object):
+    """Provide base settings."""
+
     def __init__(self, parent):
         self._parent = parent
 
     @property
     def sim_setup_info(self):
         """EDB internal simulation setup object."""
-        return self._parent._edb_sim_setup_info
+
+        return self._parent.get_sim_setup_info
+
+    @pyaedt_function_handler
+    def get_configurations(self):
+        """Get all attributes.
```

Comment:
The method says configurations yet the description says attributes. They seem like they should use the same term.
A description is needed for the "Returns" section.